public/app/features/auth-config/ProviderConfigForm.tsx

```diff
@@ -114,95 +138,106 @@ export const ProviderConfigForm = ({ config, provider, isLoading }: ProviderConf
+          onDiscard={() => {
+            reportInteraction('grafana_authentication_ssosettings_abandoned', {
+              provider,
+            });
+            reset();
+          }}
+        />
+        <Field label="Enabled" hidden={true}>
+          <Switch {...register('enabled')} id="enabled" label={'Enabled'} />
+        </Field>
+        {sections ? (
+          <Stack gap={2} direction={'column'}>
+            {sections
+              .filter((section) => !section.hidden)
+              .map((section, index) => {
                 return (
-                  <FieldRenderer
-                    key={field}
-                    field={field}
-                    control={control}
-                    errors={errors}
-                    setValue={setValue}
-                    register={register}
-                    watch={watch}
-                    unregister={unregister}
-                    provider={provider}
-                    secretConfigured={!!config?.settings.clientSecret}
-                  />
+                  <CollapsableSection label={section.name} isOpen={index === 0} key={section.name}>
+                    {section.fields
+                      .filter((field) => (typeof field !== 'string' ? !field.hidden : true))
+                      .map((field) => {
+                        return (
+                          <FieldRenderer
+                            key={typeof field === 'string' ? field : field.name}
+                            field={field}
+                            control={control}
+                            errors={errors}
+                            setValue={setValue}
+                            register={register}
+                            watch={watch}
+                            unregister={unregister}
+                            provider={provider}
+                            secretConfigured={!!config?.settings.clientSecret}
+                          />
+                        );
+                      })}
+                  </CollapsableSection>
                 );
               })}
-            </>
-          )}
-          <Box display={'flex'} gap={2} marginTop={6}>
-            <Field>
-              <Button type={'submit'} disabled={isSaving}>
+          </Stack>
+        ) : (
+          <>
+            {providerFields.map((field) => {
+              return (
+                <FieldRenderer
+                  key={field}
+                  field={field}
+                  control={control}
+                  errors={errors}
+                  setValue={setValue}
+                  register={register}
+                  watch={watch}
+                  unregister={unregister}
+                  provider={provider}
+                  secretConfigured={!!config?.settings.clientSecret}
+                />
+              );
+            })}
+          </>
+        )}
+        <Box display={'flex'} gap={2} marginTop={6}>
+          <Field>
```

Comment:
I think you can remove all these Field components for the buttons and wrap them into Box and Stack:
```tsx
 <Box display={'flex'} gap={2} marginTop={5}>
  <Stack alignItems={'center'} gap={2}>
    <Button
      type={'submit'}
      disabled={isSaving}
      onClick={() => setValue('enabled', !isEnabled)}
      variant={isEnabled ? 'secondary' : undefined}
    >
      {isSaving ? (isEnabled ? 'Disabling...' : 'Saving...') : isEnabled ? 'Disable' : 'Save and enable'}
    </Button>

    <Button type={'submit'} disabled={isSaving} variant={'secondary'}>
      {isSaving ? 'Saving...' : 'Save'}
    </Button>
    <LinkButton href={'/admin/authentication'} variant={'secondary'}>
      Discard
    </LinkButton>

    <Dropdown overlay={additionalActionsMenu} placement="bottom-start">
      <IconButton
        tooltip="More actions"
        title="More actions"
        tooltipPlacement="top"
        size="md"
        variant="secondary"
        name="ellipsis-v"
        hidden={config?.source !== 'system'}
      />
    </Dropdown>
  </Stack>
</Box>
```